py/jupyterlite-core/jupyterlite_core/addons/contents.py

```diff
@@ -156,7 +156,14 @@ def one_contents_path(self, output_file_dir, api_path):
 
         self.maybe_timestamp(self.output_files_dir)
 
-        from jupyter_server.services.contents.filemanager import FileContentsManager
+        try:
+            from jupyter_server.services.contents.filemanager import FileContentsManager
+        except ImportError as e:
+            raise RuntimeError(
+                """jupyter-server is not installed. You cannot add custom content to jupyterlite.
```

Comment:
Maybe this could / should be raised from the check above directly?

https://github.com/jupyterlite/jupyterlite/blob/a9dc3206019f40dcff913dcdcabe64b0cdcfa4f1/py/jupyterlite-core/jupyterlite_core/addons/contents.py#L148-L152